### PR TITLE
feature layer selection sample

### PR DESCRIPTION
**Pull request by apurvag123**

### CODE REVIEW

File: src/iOS/Xamarin.iOS/Samples/Features/FeatureLayerSelection/FeatureLayerSelection.cs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        }
+
+        public override void ViewDidLoad()
+        {
+            base.ViewDidLoad();
+            // Create the UI, setup the control references and execute initialization 
+            CreateLayout();
+            Initialize();
+        }
+        private async void Initialize()
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/Features/FeatureLayerSelection/FeatureLayerSelection.cs]: Please add empty lines before and after methods

File: src/iOS/Xamarin.iOS/Samples/Features/FeatureLayerSelection/FeatureLayerSelection.cs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        {
+            // Create new Map with basemap
+            var myMap = new Map(Basemap.CreateTopographic());
+
+            //set the initial viewpoint for map
+            myMap.InitialViewpoint = new Viewpoint(new Envelope(-1131596.019761, 3893114.069099, 3926705.982140, 7977912.461790, SpatialReferences.WebMercator));
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/Features/FeatureLayerSelection/FeatureLayerSelection.cs]: Split into multiple lines. Create Envelope separate from the viewpoint.

File: src/iOS/Xamarin.iOS/Samples/Features/FeatureLayerSelection/FeatureLayerSelection.cs [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+            // Provide used Map to the MapView
+            _myMapView.Map = myMap;
+
+            //initialize feature table using a url to feature server url
+            var featureTable = new ServiceFeatureTable(new Uri("http://sampleserver6.arcgisonline.com/arcgis/rest/services/DamageAssessment/FeatureServer/0"));
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/Features/FeatureLayerSelection/FeatureLayerSelection.cs]: Split into multiple lines. Create Uri separate from table.

File: src/iOS/Xamarin.iOS/Samples/Features/FeatureLayerSelection/FeatureLayerSelection.cs [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+            // Create new Map with basemap
+            var myMap = new Map(Basemap.CreateTopographic());
+
+            //set the initial viewpoint for map
+            myMap.InitialViewpoint = new Viewpoint(new Envelope(-1131596.019761, 3893114.069099, 3926705.982140, 7977912.461790, SpatialReferences.WebMercator));
+
+            // Provide used Map to the MapView
+            _myMapView.Map = myMap;
+
+            //initialize feature table using a url to feature server url
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/Features/FeatureLayerSelection/FeatureLayerSelection.cs]: Make sure that all comments follows same format
// Capital and then rest...

File: src/iOS/Xamarin.iOS/Samples/Features/FeatureLayerSelection/FeatureLayerSelection.cs [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+            await _featureLayer.LoadAsync();
+
+            //check for the load status. If the layer is loaded then add it to map
+            if (_featureLayer.LoadStatus == Esri.ArcGISRuntime.LoadStatus.Loaded)
+            {
+                //add the feature layer to the map
+                myMap.OperationalLayers.Add(_featureLayer);
+            }
+        }
+        private void CreateLayout()
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/Features/FeatureLayerSelection/FeatureLayerSelection.cs]: Move last.

File: src/iOS/Xamarin.iOS/Samples/Features/FeatureLayerSelection/FeatureLayerSelection.cs [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+        private async void OnMapViewTapped(object sender, GeoViewInputEventArgs e)
+        {
+            //define the selection tolerance
+            double tolerance = 5;
+
+            //convert the tolerance to map units
+            double mapTolerance = tolerance * _myMapView.UnitsPerPixel;
+
+            //define the envelope around the tap location for selecting features
+            var selectionEnvelope = new Envelope(e.Location.X - mapTolerance, e.Location.Y - mapTolerance, e.Location.X + mapTolerance, e.Location.Y + mapTolerance, _myMapView.Map.SpatialReference);
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/Features/FeatureLayerSelection/FeatureLayerSelection.cs]: Format to multiple lines

File: src/iOS/Xamarin.iOS/Samples/Features/FeatureLayerSelection/FeatureLayerSelection.cs [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+            View.AddSubviews(_myMapView);
+        }
+
+        private async void OnMapViewTapped(object sender, GeoViewInputEventArgs e)
+        {
+            //define the selection tolerance
+            double tolerance = 5;
+
+            //convert the tolerance to map units
+            double mapTolerance = tolerance * _myMapView.UnitsPerPixel;
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/Features/FeatureLayerSelection/FeatureLayerSelection.cs]: How does this differ from normal tolerance? Please add more information why this is done.

File: src/iOS/Xamarin.iOS/Samples/Features/FeatureLayerSelection/metadata.json [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+  "SampleName": "FeatureLayerSelection",
+  "Description": "This sample demonstrates how to select features in a feature layer by tapping a mapview.",
+  "Instructions": "",
+  "Type": 0,
+  "RequiresOnlineConnection":true,
+  "RequiresOfflineData":false,
+  "RequiresLocalServer":false,
+  "Image":"FeatureLayerSelection.jpg",
+  "Link":"",
+  "TypeLink": ["M:Esri.ArcGISRuntime.Mapping.Basemap.CreateTopographic","M:Esri.ArcGISRuntime.Mapping.Basemap.CreateStreets","M:Esri.ArcGISRuntime.Mapping.Basemap.CreateImagery","M:Esri.ArcGISRuntime.Mapping.Basemap.CreateOceans"
```
Comment (anttikajanus) [inline src/iOS/Xamarin.iOS/Samples/Features/FeatureLayerSelection/metadata.json]: Clear type links that aren't used

**Comment by anttikajanus** [comment]
Mainly some styling issues. Please fix and merge.

**Comment by apurvag123** [comment]
Made all the changes

**Comment by anttikajanus** [comment]
@apurvag123  Since changes are done, please merge the changes.

**Pull request closed without merging**